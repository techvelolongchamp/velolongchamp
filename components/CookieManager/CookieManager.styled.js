import styled from 'styled-components'

export const CookieContainer = styled.div`
  background: ${({ theme }) => theme.colors.defaultColor};
  bottom: ${({ showCookie }) => (showCookie ? '0' : '-200px')};
  transition: all 1s ease;
  color: white;
  font-size: ${({ theme }) => theme.fonts.medium};
  padding: ${({ theme }) => theme.spacing.s};
  position: fixed;
  text-align: center;
  width: 100vw;
  z-index: 1000;

  div {
    margin-top: ${({ theme }) => theme.spacing.s};
  }
  a {
    background: ${({ theme }) => theme.colors.brandSecondary};
    border-radius: 5px;
    color: ${({ theme }) => theme.colors.white};
    margin: 0 ${({ theme }) => theme.spacing.s};
    padding: ${({ theme }) => theme.spacing.xs};
  }
  button {
    background: ${({ theme }) => theme.colors.brandPrimary};
    border-radius: 5px;
    color: ${({ theme }) => theme.colors.white};
    margin: 0 ${({ theme }) => theme.spacing.s};
    outline: done;
    padding: ${({ theme }) => theme.spacing.xs};
    width: 40px;
  }
`
