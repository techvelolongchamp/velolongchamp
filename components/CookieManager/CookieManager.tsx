import React, { useState, useEffect } from 'react'
import Cookies from 'js-cookie'
import Link from 'next/link'
import { useIntl } from 'react-intl'

import { CookieContainer } from './CookieManager.styled'

const CookieManager = () => {
  const { formatMessage } = useIntl()
  const [cookieApproval, setCookieApproval] = useState(true)

  useEffect(() => {
    const cookieValue = Boolean(Cookies.get('cookie-approval'))
    if (cookieValue !== cookieApproval) setCookieApproval(cookieValue)
  }, [])

  const handleClick = (e: React.MouseEvent) => {
    e.preventDefault()
    Cookies.set('cookie-approval', 'true')
    setCookieApproval(true)
  }

  return (
    <CookieContainer showCookie={!cookieApproval}>
      <p>{formatMessage({ id: 'cookie.message' })}</p>
      <div>
        <button onClick={handleClick}>
          {formatMessage({ id: 'cookie.ok' })}
        </button>
        <Link href="/legal">{formatMessage({ id: 'cookie.learnMore' })}</Link>
      </div>
    </CookieContainer>
  )
}

export default CookieManager
