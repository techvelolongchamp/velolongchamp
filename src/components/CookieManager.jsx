import React, { useState } from 'react'
import Cookies from 'js-cookie'
import styled from 'styled-components'
import { Link } from 'gatsby'

const CookieContainer = styled.div`
  background: ${({ theme }) => theme.colors.defaultColor};
  bottom: ${({ showCookie }) => (showCookie ? '0' : '-200px')};
  transition: all 1s ease;
  color: white;
  font-size: ${({ theme }) => theme.fonts.medium};
  padding: ${({ theme }) => theme.spacing.s};
  position: fixed;
  text-align: center;
  width: 100vw;
  z-index: 1000;

  div {
    margin-top: ${({ theme }) => theme.spacing.s};
  }
  a {
    background: ${({ theme }) => theme.colors.brandSecondary};
    border-radius: 5px;
    color: ${({ theme }) => theme.colors.white};
    margin: 0 ${({ theme }) => theme.spacing.s};
    padding: ${({ theme }) => theme.spacing.xs};
  }
  button {
    background: ${({ theme }) => theme.colors.brandPrimary};
    border-radius: 5px;
    color: ${({ theme }) => theme.colors.white};
    margin: 0 ${({ theme }) => theme.spacing.s};
    outline: done;
    padding: ${({ theme }) => theme.spacing.xs};
    width: 40px;
  }
`

const CookieManager = () => {
  const [cookieApproval, setCookieApproval] = useState(
    Boolean(Cookies.get('cookie-approval'))
  )
  const handleClick = e => {
    e.preventDefault()
    Cookies.set('cookie-approval', true)
    setCookieApproval(true)
  }
  return (
    <CookieContainer showCookie={!cookieApproval}>
      <p>
        En poursuivant votre navigation sur notre site, vous acceptez
        l'utilisation de cookies afin de nous permettre d'améliorer votre
        expérience utilisateur.
      </p>
      <div>
        <button onClick={handleClick}>Ok</button>
        <Link to="/legal">En savoir plus</Link>
      </div>
    </CookieContainer>
  )
}

export default CookieManager
