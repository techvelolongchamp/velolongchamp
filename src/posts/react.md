---
title: "React"
date: "2019-05-02"
---

I started to learn React end of 2016. Since then I have improved and developed various skills thanks to my experiences and side projects.

## Key skills

1. React
2. React Hooks
3. Redux / Rematch
4. React Native
5. GraphQL
6. Nextjs
7. Gatsby
8. Jest
9. Styled components

*This post is generated from a markdown file*