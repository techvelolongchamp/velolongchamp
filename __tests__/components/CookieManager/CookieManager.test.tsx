import React from 'react'
import { act, fireEvent } from '@testing-library/react'

import { RenderWithProvider as render } from '../../provider'

import CookieManager from '../../../components/CookieManager/CookieManager'

describe('CookieManager unit tests', () => {
  it('should render correctly', () => {
    const { container } = render(<CookieManager />)

    expect(container).toMatchSnapshot()
  })

  it('should render correctly - cookie accepted', () => {
    const { container, getByTestId } = render(<CookieManager />)
    const accept = getByTestId('accept-cookie')
    fireEvent.click(accept)
    act(() => expect(container).toMatchSnapshot())
  })
})
