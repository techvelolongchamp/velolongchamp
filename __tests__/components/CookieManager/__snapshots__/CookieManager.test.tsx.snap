// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieManager unit tests should render correctly - cookie accepted 1`] = `
.c0 {
  background: #0d171b;
  bottom: -200px;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
  color: white;
  font-size: 14px;
  padding: 10px;
  position: fixed;
  text-align: center;
  width: 100vw;
  z-index: 1000;
}

.c0 div {
  margin-top: 10px;
}

.c0 a {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  margin: 0 10px;
  padding: 5px;
}

.c0 button {
  background: #419a1c;
  border-radius: 5px;
  color: #fff;
  margin: 0 10px;
  outline: done;
  padding: 5px;
  width: 40px;
}

<div>
  <div
    class="c0"
  >
    <p>
      En poursuivant votre navigation sur notre site, vous acceptez l'utilisation de cookies afin de nous permettre d'améliorer votre expérience utilisateur.
    </p>
    <div>
      <button
        data-testid="accept-cookie"
      >
        OK
      </button>
      <a
        href="/legal"
      >
        En savoir plus
      </a>
    </div>
  </div>
</div>
`;

exports[`CookieManager unit tests should render correctly 1`] = `
.c0 {
  background: #0d171b;
  bottom: 0;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
  color: white;
  font-size: 14px;
  padding: 10px;
  position: fixed;
  text-align: center;
  width: 100vw;
  z-index: 1000;
}

.c0 div {
  margin-top: 10px;
}

.c0 a {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  margin: 0 10px;
  padding: 5px;
}

.c0 button {
  background: #419a1c;
  border-radius: 5px;
  color: #fff;
  margin: 0 10px;
  outline: done;
  padding: 5px;
  width: 40px;
}

<div>
  <div
    class="c0"
  >
    <p>
      En poursuivant votre navigation sur notre site, vous acceptez l'utilisation de cookies afin de nous permettre d'améliorer votre expérience utilisateur.
    </p>
    <div>
      <button
        data-testid="accept-cookie"
      >
        OK
      </button>
      <a
        href="/legal"
      >
        En savoir plus
      </a>
    </div>
  </div>
</div>
`;
